pandapipes/control/run_control.py

```diff
@@ -35,10 +35,19 @@ def prepare_run_ctrl(net, ctrl_variables):
     :return: ctrl_variables
     :rtype: dict
     """
+    ctrl_var = ctrl_variables
+
     if ctrl_variables is None:
         ctrl_variables  = prepare_run_control_pandapower(net, None)
         ctrl_variables["run"] = ppipe.pipeflow
 
+    if ('continue_on_divergence') in kwargs and (ctrl_var is None):
+        div = kwargs.pop('continue_on_divergence')
+        ctrl_variables['continue_on_divergence'] = div
+    if ('check_each_level') in kwargs and (ctrl_var is None):
```

Comment:
If the if in L44 was True, this will always be False because ctrl_var is not None.
Like above, I'd expect something like `and ((ctrl_var is None) or (continue_on_divergence not in ctrl_var.keys()))`